ext/ffi/lib.rs

```diff
@@ -535,322 +564,792 @@ where
-      } as u64))
+    FFI_TYPE_UINT16 => {
+      *(result as *mut u16) = value
+        .number_value(&mut scope)
+        .expect("Unable to deserialize result parameter.")
+        as u16;
     }
-    NativeType::USize => {
-      json!(U32x2::from(unsafe {
-        symbol.cif.call::<usize>(symbol.ptr, &call_args)
-      } as u64))
+    FFI_TYPE_UINT32 => {
+      *(result as *mut u32) = value
+        .number_value(&mut scope)
+        .expect("Unable to deserialize result parameter.")
+        as u32;
     }
-    NativeType::ISize => {
-      json!(U32x2::from(unsafe {
-        symbol.cif.call::<isize>(symbol.ptr, &call_args)
-      } as u64))
+    FFI_TYPE_SINT64 => {
+      if value.is_big_int() {
+        let value = v8::Local::<v8::BigInt>::try_from(value).unwrap();
+        *(result as *mut i64) = value.i64_value().0;
+      } else {
+        *(result as *mut i64) = value
+          .number_value(&mut scope)
+          .expect("Unable to deserialize result parameter.")
+          as i64;
+      }
     }
-    NativeType::F32 => {
-      json!(unsafe { symbol.cif.call::<f32>(symbol.ptr, &call_args) })
+    FFI_TYPE_UINT64 => {
+      if value.is_big_int() {
+        let value = v8::Local::<v8::BigInt>::try_from(value).unwrap();
+        *(result as *mut u64) = value.u64_value().0;
+      } else {
+        *(result as *mut u64) = value
+          .number_value(&mut scope)
+          .expect("Unable to deserialize result parameter.")
+          as u64;
+      }
     }
-    NativeType::F64 => {
-      json!(unsafe { symbol.cif.call::<f64>(symbol.ptr, &call_args) })
+    FFI_TYPE_VOID => {
+      // nop
     }
-    NativeType::Pointer => {
-      json!(U32x2::from(unsafe {
-        symbol.cif.call::<*const u8>(symbol.ptr, &call_args)
-      } as u64))
+    _ => {
+      unreachable!();
     }
-  })
+  };
+}
+
+#[derive(Deserialize)]
+struct RegisterCallbackArgs {
+  parameters: Vec<NativeType>,
+  result: NativeType,
+}
+
+#[op(v8)]
+fn op_ffi_register_callback(
+  state: &mut deno_core::OpState,
+  scope: &mut v8::HandleScope,
+  args: RegisterCallbackArgs,
+  cb: serde_v8::Value<'_>,
+) -> Result<(ResourceId, U32x2), AnyError> {
+  let v8_value = cb.v8_value;
+  let cb = v8::Local::<v8::Function>::try_from(v8_value)?;
+
+  let info = {
+    let isolate_ptr: *mut v8::Isolate = {
+      let isolate: &mut v8::Isolate = &mut *scope;
+      isolate
+    };
+    let isolate = unsafe { &mut *isolate_ptr };
+    let callback = v8::Global::new(isolate, cb).into_raw();
+    let context =
+      v8::Global::new(isolate, scope.get_current_context()).into_raw();
+
+    Box::leak(Box::new(CallbackInfo {
+      callback,
+      context,
+      isolate,
+    }))
+  };
+  let cif = Cif::new(
+    args.parameters.into_iter().map(libffi::middle::Type::from),
+    libffi::middle::Type::from(args.result),
+  );
+
+  let closure = libffi::middle::Closure::new(cif, deno_ffi_callback, info);
+  let u3x2 = U32x2::from(*closure.code_ptr() as usize as u64);
```

Comment:
Add a TODO to use BigInt later.

Reply:
I could also change this code to manually create a `v8::Array` and add the two items?